### PR TITLE
Docker updates and a few fixes for Serval

### PR DESCRIPTION
**Pull request by johnml1135**
New things:L
* Create docker-dev environment for debugging in vscode in a docker container
  * Add sample file that calls machine.py as serval/machine calls it
* Make saving models optional, and defaulted to on - to preserve existing behaviour
Fix a few issues:
* Fix poetry dependency in typing-extensions and python v3.8
* Fix  PretranslationInfo parameter from "segment" to "translation" to align with updates from machine and update tests
* Update poetry version to fix bug - https://github.com/python-poetry/poetry/issues/7211

- - -
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/machine.py/11)

### CODE REVIEW

**Comment by johnml1135** [comment]
Also, updated black to the most recent version, added the vscode formatter and ran black on all python files.

File: machine/webapi/clearml_nmt_engine_build_job.py [hunk 1]
```diff
@@ -83,8 +83,11 @@ def run(self, check_canceled: Optional[Callable[[], None]] = None) -> None:
                     check_canceled()
                 _translate_batch(model, pi_batch, source_tokenizer, target_detokenizer, writer)
 
-        print("Saving NMT model")
-        self._nmt_model_factory.save_model()
+        if self._config.save_model:
```
Comment (ddaspit) [inline machine/webapi/clearml_nmt_engine_build_job.py:86]: I am planning on removing this line of code, so this isn't necessary.

**Review by ddaspit — CHANGES_REQUESTED** [review]

File: machine/webapi/clearml_nmt_engine_build_job.py (hunk 1 above)
Comment (johnml1135) [reply machine/webapi/clearml_nmt_engine_build_job.py:86]: I'll tear it out now and we can track the need to add it in here: #12.

**Review by ddaspit — APPROVED** [review]
:lgtm:

> Reviewed 13 of 16 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/machine.py/11#-:-NVgv5npDF8M6pIPd-yc:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @johnml1135)

**Pull request merged**